api/index.js

```diff
@@ -93,10 +114,91 @@ class Ekke extends EventEmitter {
     this.on('error', args => console.error(...args));
 
     //
-    // Generic event handling
+    // Generic event handling.
     //
     this.on('ping', (args, send) => send('pong', args));
   }
+
+  /**
+   * Execute all assigned functions for the given map & method.
+   *
+   * @param {String} name Name of the plugin map we should run on.
+   * @param {String} method Name of the method we want to execute.
+   * @param {Object} data Data the functions should receive.
+   * @public
+   */
+  async exec(name, method, data = {}, ...args) {
+    const map = this.plugins[name];
+    const methods = map.get(method);
+    let result = data;
+
+    if (!methods) return result;
+
+    for (let i = 0; i < methods.length; i++) {
+      const fn = methods[i].fn;
+      const name = fn.displayName || fn.name;
+      const timeout = ms(fn.timeout || '20 seconds');
+
+      try {
+        result = await borked(fn(result, ...args), timeout) || result;
+      } catch (e) {
+        debug(`failed to execute(${name})`, e);
```

Comment:
should this eat all exceptions? and allow it to continue?  Based on some of your usages, it may not be safe to do that (e.g. babel & metro config later in this PR)

Reply:
Yes, you are correct. I have already fixed this locally by adding a `this.emit('error', e)` so these errors do not disappear into the void.